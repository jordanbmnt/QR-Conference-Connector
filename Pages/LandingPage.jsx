import { useSelector } from "react-redux";
import { Box, GluestackUIProvider } from "@gluestack-ui/themed";
import { config } from "@gluestack-ui/config";
import ScannerContainer from "../Components/ScannerContainer";
import ContactInfoPage from "./ContactInfoPage";

export default function LandingPage() {
  const view = useSelector((state) => state.appSlice.viewContactInfo);
  const contactStateInfo = useSelector((state) => state.appSlice.contactInfo);
  return (
    <GluestackUIProvider config={config}>
      <Box
        style={{ height: "100%" }}
        contentContainerStyle={{ flexGrow: 1 }}
        backgroundColor='#0b0e14'
      >
        {!view ? (
          <ScannerContainer />
        ) : (
          <ContactInfoPage info={contactStateInfo} />
        )}
      </Box>
    </GluestackUIProvider>
  );
}
