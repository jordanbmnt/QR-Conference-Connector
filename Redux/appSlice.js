import { createSlice } from '@reduxjs/toolkit';

export const appSlice = createSlice({
  name: 'appSlice',
  initialState: {
    viewContactInfo: false,
    scanViewVisible: true,
    previousPage: false,
    contactInfo: {},
    alertInfo: {
      visibility: false,
      theme: "",
    }
  },
  reducers: {
    setViewContact: (state, action) => {
      state.viewContactInfo = action.payload;
    },
    setScanViewVisible: (state, action) => {
      state.scanViewVisible = action.payload;
    },
    setPreviousPage: (state, action) => {
      state.previousPage = action.payload;
    },
    setContactInfo: (state, action) => {
      state.contactInfo = action.payload;
    },
    setAlertVisibility: (state, action) => {
      state.alertInfo.visibility = action.payload;
    },
    setAlertTheme: (state, action) => {
      state.alertInfo.theme = action.payload;
    },
  },
});

export const { setViewContact, setContactInfo, setScanViewVisible, setPreviousPage, setAlertTheme, setAlertVisibility } = appSlice.actions;


export default appSlice.reducer;
