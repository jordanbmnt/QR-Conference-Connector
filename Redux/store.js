import { configureStore } from '@reduxjs/toolkit';
import rootReducer from './reducers'

export const setupUserInfoStore = (preloadedState) => {
  return configureStore({
    reducer: rootReducer,
    preloadedState,
  });
};

export const store = configureStore({
  reducer: rootReducer
})