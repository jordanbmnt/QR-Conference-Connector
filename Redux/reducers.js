import { combineReducers } from 'redux';
import  appSlice  from './appSlice';

export default combineReducers({ appSlice })