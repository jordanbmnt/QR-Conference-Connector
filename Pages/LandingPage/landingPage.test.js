import LandingPage from "./LandingPage";
import renderer, { act } from "react-test-renderer";
import { Provider } from "react-redux";
import { GluestackUIStyledProvider } from "@gluestack-ui/themed";
import { config } from "@gluestack-ui/config";
import { setupUserInfoStore } from "../../Redux/store";

function MockBarCodeScanner() {
  return <></>;
}

MockBarCodeScanner.requestPermissionsAsync = async function () {
  return { status: "granted" };
};

jest.mock("expo-barcode-scanner", () => ({
  __esModule: true,
  BarCodeScanner: MockBarCodeScanner,
}));

jest.mock("expo-sqlite", () => ({
  openDatabase: jest.fn(() => ({
    transaction: jest.fn(),
  })),
}));

describe("LandingPage", function () {
  let landingPage;
  const initialState = {
    appSlice: {
    viewContactInfo: false,
      contactInfo: {},
    },
  };

  beforeAll(async () => {
    const store = setupUserInfoStore(initialState);
    await act(async () => {
      landingPage = renderer.create(
        <Provider store={store}>
          <GluestackUIStyledProvider config={config}>
            <LandingPage />
          </GluestackUIStyledProvider>
        </Provider>
      );
    });
  });

  it("should render LandingPage component correctly", () => {
    act(() => {
      expect(landingPage).toMatchSnapshot();
    });
  });

  it("should have a single child", function () {
    expect(landingPage.toJSON().children.length).toBe(1);
  });
});
