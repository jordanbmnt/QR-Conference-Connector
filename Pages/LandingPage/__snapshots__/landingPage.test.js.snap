// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should render LandingPage component correctly 1`] = `
<View
  contentContainerStyle={
    {
      "flexGrow": 1,
    }
  }
  data-style=""
  dataSet={
    {
      "style": "",
    }
  }
  style={
    [
      undefined,
      {},
      {},
      {},
      {
        "backgroundColor": "#0b0e14",
      },
      {
        "height": "100%",
      },
    ]
  }
>
  <View
    data-style=""
    dataSet={
      {
        "style": "",
      }
    }
    style={
      [
        undefined,
        {},
        {},
        {},
        {
          "alignItems": "center",
          "borderColor": "#ffedd5",
          "borderRadius": 8,
          "borderWidth": 2,
          "height": "60%",
          "justifyContent": "space-between",
        },
        {
          "height": "80%",
          "marginHorizontal": 20,
          "marginVertical": 64,
        },
        undefined,
      ]
    }
    sx={
      {
        "@base": {
          "height": "80%",
          "mx": "$5",
          "my": "$16",
        },
        "@lg": {
          "mx": "$32",
          "my": "$24",
        },
        "borderColor": "$orange100",
        "borderWidth": "$2",
      }
    }
  >
    <View
      style={
        {
          "alignContent": "center",
          "height": "80%",
          "justifyContent": "center",
          "opacity": 0.5,
        }
      }
    >
      <Image
        alt="noContact"
        source={1}
        style={
          {
            "height": 400,
            "width": 400,
          }
        }
      />
    </View>
    <View
      data-style=""
      dataSet={
        {
          "style": "",
        }
      }
      style={
        [
          undefined,
          {},
          {
            "flexDirection": "row",
          },
          {},
          undefined,
        ]
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": false,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        action="primary"
        collapsable={false}
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        role="button"
        size="md"
        style={
          [
            undefined,
            {},
            {
              "alignItems": "center",
              "backgroundColor": "#0077E6",
              "borderRadius": 4,
              "flexDirection": "row",
              "justifyContent": "center",
            },
            {},
            {
              "backgroundColor": "#0077E6",
              "borderColor": "#4AA9FF",
            },
            {},
            {
              "height": 40,
              "paddingHorizontal": 20,
            },
            {},
            {
              "backgroundColor": "#0077E6",
              "borderColor": "#4AA9FF",
            },
            {},
            {
              "height": 40,
              "paddingHorizontal": 20,
            },
            {},
            {
              "margin": 30,
              "marginBottom": 40,
              "marginTop": 0,
            },
            undefined,
          ]
        }
        sx={
          {
            "margin": 30,
            "marginBottom": 40,
            "marginTop": 0,
          }
        }
        variant="solid"
      >
        <Text
          data-style=""
          dataSet={
            {
              "style": "",
            }
          }
          size="md"
          style={
            [
              undefined,
              {},
              {
                "color": "#525252",
                "fontFamily": undefined,
                "fontStyle": "normal",
                "fontWeight": "400",
                "letterSpacing": 0,
              },
              {},
              {
                "fontSize": 16,
                "lineHeight": 22,
              },
              undefined,
              {},
              {
                "color": "#FCFCFC",
              },
              {},
              {
                "color": "#FCFCFC",
                "fontWeight": "600",
              },
              {
                "color": "#005DB4",
              },
              {
                "color": "#FCFCFC",
              },
              {},
              {
                "color": "#FCFCFC",
              },
              undefined,
            ]
          }
        >
          Tap to Scan Again
        </Text>
      </View>
    </View>
  </View>
</View>
`;
